src/libprojectM/Renderer/BeatDetect.cpp

```diff
@@ -104,60 +104,36 @@ void BeatDetect::detectFromSamples()
     treb=0;
     vol=0;
 
-    // TODO: get sample rate from PCM?  Assume 44100
-    getBeatVals(44100.0f, FFT_LENGTH, pcm->pcmdataL, pcm->pcmdataR);
+    float vdataL[FFT_LENGTH];
+    float vdataR[FFT_LENGTH];
+    pcm->getSpectrum(vdataL, 0, FFT_LENGTH, 0.0);
+    pcm->getSpectrum(vdataR, 1, FFT_LENGTH, 0.0);
+    getBeatVals(44100.0f, FFT_LENGTH, vdataL, vdataR);
```

Comment:
We really shouldn't assume 44.1k here; the input sample rate can vary. At least leave the TODO here?

Reply:
added comment